database/engine/rrdenginelib.h

```diff
@@ -58,7 +58,7 @@ static inline void modify_bit(unsigned *x, unsigned pos, uint8_t val)
     }
 }
 
-#define RRDENG_PATH_MAX (4096)
+#define RRDENG_PATH_MAX (4096+1)
```

Comment:
Considering that we are talking about `PATH_MAX`, I suggest we use `FILENAME_MAX + 1`.